src/dishka/registry.py

```diff
@@ -320,12 +321,12 @@ def _process_decorator(
             )
         scope = self.dependency_scopes[provides]
         registry = self.registries[scope]
-        undecorated_type = NewType(
+        undecorated_type = NewType(  # type: ignore[misc]
```

Comment:
why?

Reply:
error: Argument 1 to NewType(...) must be a string literal

Reply:
Looks like we need validate if it None here.